File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/domain/model/equipment/EquipmentId.java
package org.taonaw.studio_reservation.domain.model.equipment;

import de.huxhorn.sulky.ulid.ULID;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import org.taonaw.studio_reservation.domain.shared.Assertion;

@Getter
@EqualsAndHashCode
public class EquipmentId {
    private final String value;

    public EquipmentId(String value) {
        Assertion.required(value);
        this.value = value;
    }

    public static EquipmentId newId() {
        return new EquipmentId(new ULID().nextULID());
    }
}

File: src/backend/facility/src/main/kotlin/org/taonaw/facility/domain/model/equipment/EquipmentId.kt
package org.taonaw.facility.domain.model.equipment

import de.huxhorn.sulky.ulid.ULID

data class EquipmentId(val value: String) {
    companion object {
        fun newId(): EquipmentId {
            return EquipmentId(ULID().nextULID())
        }
    }
}